### PR TITLE
Use mongodb driver types for Selector=>Filter, Modifier=>UpdateFilter, SortSpecifier=>Sort instead of homebuilt types

### PR DESCRIPTION
**Pull request by perbergland**
Now that mongodb has great typescript support, bring in filter, updatefilter and sort types from the driver types and remove the now unused old types.

Replica of https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63759

### CODE REVIEW

**Comment by perbergland** [comment]
also copied over a fix for the cursor iterators, since the current code made it unusable

**Review by StorytellerCZ — APPROVED** [review]

File: packages/mongo/mongo.d.ts [hunk 1]
```diff
@@ -15,78 +15,11 @@ export type UnionOmit<T, K extends keyof any> = T extends T
...
-  type Query<T> = {
-    [P in keyof T]?: Flatten<T[P]> | RegExp | FieldExpression<Flatten<T[P]>>;
-  } & {
-    $or?: Query<T>[] | undefined;
-    $and?: Query<T>[] | undefined;
-    $nor?: Query<T>[] | undefined;
-  } & Dictionary<any>;
+  /**
+   * Alias for {@link MongoNpmModule.Filter}
+   */
```
Comment (radekmie) [inline packages/mongo/mongo.d.ts:21]: I think this comment is not needed. (Or other types should get one too.)

**Review by radekmie — APPROVED** [review]

**Comment by perbergland** [comment]
I also felt like adding the missing (new async methods in 2.9.1) countDocuments and estimatedDocumentCount since I couldn’t see any PR or issue for adding them, but it felt like scope creep. Whaddya think.

File: packages/mongo/mongo.d.ts (hunk 1 above)
Comment (perbergland) [reply packages/mongo/mongo.d.ts]: Removed it (along with a dozen other now unused types)

**Comment by Grubba27** [comment]
> Whaddya think?

Sounds great! 2.9.1 is already out, but I'm thinking of making a 2.9.2 with a few small additions. I could squeeze this one there I think

**Pull request merged**